hle-survey/transformation/src/main/scala/org/broadinstitute/monster/dap/RawRecord.scala

```diff
@@ -43,6 +43,13 @@ case class RawRecord(id: Long, fields: Map[String, Array[String]]) {
   /** Get the singleton value for an attribute in this record if one exists, parsed as a boolean. */
   def getOptionalBoolean(field: String): Option[Boolean] = getOptional(field).map(_ == "1")
 
+  /** Get the singleton value for an attribute in this record, parsed as a long. */
+  def getOptionalNumber(field: String): Option[Long] = getOptional(field).map(_.toLong)
+
   /** Get every value for an attribute in this record. */
   def getArray(field: String): Array[String] = fields.getOrElse(field, Array.empty)
+
+  /** If the field exists, check whether it contains the value, otherwise return None. */
+  def containsValue(field: String, value: String): Option[Boolean] =
```

Comment:
I'm not sure about the utility of this helper. It runs the `getOrElse` per-call, and forces `Option[Boolean]` (which will be the most common case, but if we ever need `Boolean` we'll need to redo it).

I'd rather have the mapping code use `getArray` to pull the values once, then call `contains` for each option.

Reply:
Okay, sounds good. I agree that it's a bit redundant. We do want it to return `None` when the field is not defined though, right? It looks like doing `getArray(...).contains(...)` would return false when the field is not defined in the RawRecord.

Reply:
To clarify, I think this method should go away entirely (so "it" won't return anything anymore).

Whether or not the end value is wrapped in an `Option` will depend on the use-site. The simplest thing will be to wrap the values in `Some` when filling in the `case class`